src/js/profiling/convertHermesProfile.ts

```diff
@@ -0,0 +1,160 @@
+import type { FrameId, StackId, ThreadCpuFrame,ThreadCpuSample, ThreadCpuStack, ThreadId } from '@sentry/types';
+import { logger } from '@sentry/utils';
+
+import type * as Hermes from './hermes';
+import { parseHermesStackFrameName } from './hermes';
+import type { RawThreadCpuProfile } from './types';
+
+const ANONYMOUS_FUNCTION_NAME = 'anonymous';
+const UNKNOWN_STACK_ID = -1;
+const JS_THREAD_NAME = 'JavaScriptThread';
+const JS_THREAD_PRIORITY = 1;
+
+/**
+ * Converts a Hermes profile to a Sentry profile.
+ *
+ * Maps Hermes samples to Sentry samples.
+ * Maps Hermes stack frames to Sentry frames.
+ * Hermes stack frame is an object representing a function call in the stack
+ * with a link to its parent stack frame. Root of the represented stack tree
+ * is main function call in Hermes that is [root] stack frame.
+ *
+ * @returns Sentry profile or null if no samples are found.
+ */
+export function convertToSentryProfile(hermesProfile: Hermes.Profile): RawThreadCpuProfile | null {
+  if (hermesProfile.samples.length === 0) {
+    logger.warn('[Profiling] No samples found in profile.');
+    return null;
+  }
+
+  const { samples, hermesStacks, jsThreads } = mapSamples(hermesProfile.samples);
+
+  const { frames, hermesStackFrameIdToSentryFrameIdMap } = mapFrames(hermesProfile.stackFrames);
+
+  const { stacks, hermesStackToSentryStackMap } = mapStacks(
+    hermesStacks,
+    hermesProfile.stackFrames,
+    hermesStackFrameIdToSentryFrameIdMap,
+);
+
+  for (const sample of samples) {
+    sample.stack_id = hermesStackToSentryStackMap.get(sample.stack_id) ?? UNKNOWN_STACK_ID;
```

Comment:
Why do you need nullish coalescing here? Is it for safety? If you are not sure of the exact case of when it might happen then I would rather elect to throw (especially during beta phase) so that we can find cases where this happens and add better handling/comments for why we need the fallback

Reply:
I've changed it here.
https://github.com/getsentry/sentry-react-native/blob/bfe91bbf99ddaeb333e3a50121057fd3388ca7c4/src/js/profiling/convertHermesProfile.ts#L43

Reply:
I know throwing an error would be better to get to know about the issue, but it will also crash the app, so `logger.error` feels like a good middle ground.

Reply:
https://develop.sentry.dev/sdk/philosophy/#degrade-gracefully
I'd not throw either.